src/templates/plugin.routing.js

```diff
@@ -86,7 +87,8 @@ function resolveRoute (route, locale) {
       if (isPrefixed) {
         localizedRoute.path = `/${locale}${localizedRoute.path}`
       }
-      localizedRoute.path = localizedRoute.path.replace(/\/+$/, '') + (nuxtOptions.trailingSlash ? '/' : '') || '/'
+      const parsedPath = parsePath(localizedRoute.path)
+      localizedRoute.path = (nuxtOptions.trailingSlash ? withTrailingSlash(parsedPath.pathname) : withoutTrailingSlash(parsedPath.pathname)) + parsedPath.search + parsedPath.hash
```

Comment:
Looks good but I would rather delegate joining back the path to ufo's `stringifyParsedURL`.

Reply:
Do you mean this?

```javascript
const parsedPath = parsePath(localizedRoute.path)
parsedPath.pathname = nuxtOptions.trailingSlash ? withTrailingSlash(parsedPath.pathname) : withoutTrailingSlash(parsedPath.pathname)
localizedRoute.path = stringifyParsedURL(parsedPath)
```

It's adding extra `?` and basically breaks, looks like a bug https://github.com/unjs/ufo/blob/2a21bc6b821eafd2284cacdf1f99a4bd133c1fd7/src/parse.ts#L68

Thanks!

Reply:
I don't have much confidence in ufo at this point...

Reply:
I have created a new issue https://github.com/unjs/ufo/issues/28

Maybe I should take a look at another library that might do better.

Thanks!

Reply:
We don't have much choice as there is no alternative.

I have created a fix, let's see when this will be merged, I'll update this PR  with your suggested changes.

Thanks!